var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -162,7 +178,7 @@ def setup_dependent_package(self, module, dependent_spec):
     def autoreconf(self, spec, prefix):
         """Not needed usually, configure should be already there"""
         # If configure exists nothing needs to be done
-        if os.path.exists(self.configure_abs_path):
+        if os.path.exists(self.configure_abs_path) and '+hwloc' not in spec:
```

Comment:
```suggestion
        if os.path.exists(self.configure_abs_path) and 'patches=eb982de3366d48cbc55eb5e0df43373a45d9f51df208abf0835a72dc6c0b4774' not in spec:
```
This saves us a lengthy autoreconf whenever possible (not necessary with earlier unpatched versions or 3.4 and later).